### PR TITLE
HARP-12773: Roadshield icon disappear, while text remains visible

### PR DESCRIPTION
**Pull request by StefanDachwitz**
* also fixes infinite render loop due to only part of road shields rendered

Signed-off-by: stefan.dachwitz <stefan.dachwitz@here.com>

Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

File: @here/harp-mapview/lib/text/TextElementsRenderer.ts [hunk 1]
```diff
@@ -1574,6 +1574,7 @@ export class TextElementsRenderer {
     ): boolean {
         const pointLabel: TextElement = labelState.element;
         const textRenderState: RenderState | undefined = labelState.textRenderState;
+        const isRoadShield = iconIndex !== undefined;
```
Comment (FraukeF) [inline @here/harp-mapview/lib/text/TextElementsRenderer.ts:1577]: Just a minor thing: This name brings more meaning to this as the condition gives, shouldn´t it be just something like "hasIcon"

**Review by FraukeF — DISMISSED** [review]

File: @here/harp-mapview/lib/text/TextElementsRenderer.ts (hunk 1 above)
Comment (atomicsulfate) [reply @here/harp-mapview/lib/text/TextElementsRenderer.ts:1577]: This constant is to differentiate between a point label or a line marker. Better name it "isLineMarker" to use the same naming as in the rest of the code.

File: @here/harp-mapview/lib/text/TextElementsRenderer.ts [hunk 2]
```diff
@@ -1831,6 +1840,7 @@ export class TextElementsRenderer {
 
         // Process markers (with shield groups).
         if (minDistanceSqr > 0 && shieldGroup !== undefined) {
+            let numShieldsVisible = 0;
```
Comment (atomicsulfate) [inline @here/harp-mapview/lib/text/TextElementsRenderer.ts:1843]: Use numMarkersVisible instead.

**Review by atomicsulfate — CHANGES_REQUESTED** [review]
Please add a unit test in TextElementsRenderer to verify the fix.

File: @here/harp-mapview/lib/text/TextElementsRenderer.ts (hunk 1 above)
Comment (StefanDachwitz) [reply @here/harp-mapview/lib/text/TextElementsRenderer.ts]: renamed to "isLineMarker"

**Review by atomicsulfate — APPROVED** [review]

**Pull request merged**